scripts/pull_github_pr.sh

```diff
@@ -55,13 +65,11 @@ check_jenkins_job_status() {
   echo "Checking Jenkins job status for job: $jenkins_job"
 
   # Fetch Jenkins job last build status without passing credentials explicitly
-  lastCompletedBuild="${jenkins_url}/job/${jenkins_job}/lastCompletedBuild/api/json?tree=result"
-  lastCompleted=$(curl -s --user $JENKINS_USERNAME:$JENKINS_API_TOKEN $lastCompletedBuild | jq -r '.result')
-  # Handle the case where authentication fails
-  if [[ "$lastCompleted" == *"Authentication required"* ]]; then
-      echo "Failed to authenticate with Jenkins. Please check your session or credentials."
-      exit 1
+  lastCompletedBuild="${jenkins_url}/job/${jenkins_job}/lastCompletedBuild"
+  if [[ "$lastCompletedBuild" == *"Unauthorized"* ]]; then
+      echo "Failed to authenticate with Jenkins. please check your JENKINS_USERNAME and JENKINS_API_TOKEN setting"
```

Comment:
This code looks completely wrong to me - unless I'm misunderstanding something.
First you removed (by mistake?) the "exit 1".
Then, you check the URL of the request (!) instead of its result??

Please make sure you actually test (manually, I'm not expecting anything fancier) each feature you add in a scenario where it is reached!